exporter/containerimage/writer.go

```diff
@@ -226,11 +225,17 @@ func (ic *ImageWriter) commitDistributionManifest(ctx context.Context, ref cache
 		"containerd.io/gc.ref.content.0": configDigest.String(),
 	}
 
+	cs := ic.opt.ContentStore
 	for i, dp := range diffPairs {
-		info, err := ic.opt.ContentStore.Info(ctx, dp.Blobsum)
+		info, err := cs.Info(ctx, dp.Blobsum)
 		if err != nil {
 			return nil, errors.Wrapf(err, "could not find blob %s from contentstore", dp.Blobsum)
 		}
+
+		layerType, err := detectLayerMediaType(ctx, cs, dp.Blobsum, oci)
```

Comment:
Instead of calling detect in here can we have `DiffPair` contain the mediatype via descriptor. This should be possible for every cache record now via `Info()` https://github.com/moby/buildkit/blob/master/cache/refs.go#L54 , it might (or not) be missing for some migrated ones from before lease based storage but even then would be good to detect once and then keep around.